plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -0,0 +1,321 @@
+                schemaName.orElse(null),
+                escapeNamePattern(tableName, metadata.getSearchStringEscape()).orElse(null),
+                new String[] {"TABLE", "VIEW"});
+    }
+
+    @Override
+    public void renameColumn(ConnectorSession session, JdbcTableHandle handle, JdbcColumnHandle jdbcColumn,
+            String newColumnName)
+    {
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            DatabaseMetaData metadata = connection.getMetaData();
+            if (metadata.storesUpperCaseIdentifiers()) {
+                newColumnName = newColumnName.toUpperCase(ENGLISH);
+            }
+            String sql = format("ALTER TABLE %s RENAME COLUMN %s TO %s ",
+                    quoted(handle.getRemoteTableName()),
+                    quoted(jdbcColumn.getColumnName()), quoted(newColumnName));
+            execute(connection, sql);
+        }
+        catch (SQLException e) {
+            throw new TrinoException(JDBC_ERROR, e);
+        }
+    }
+
+    @Override
+    protected void renameTable(ConnectorSession session, String catalogName, String schemaName, String tableName,
+            SchemaTableName newTable)
+    {
+        String sql = format("RENAME TABLE %s.%s TO %s.%s", schemaName, tableName, newTable.getSchemaName(),
+                newTable.getTableName());
+        execute(session, sql);
+    }
+
+    @Override
+    public boolean isLimitGuaranteed(ConnectorSession session)
+    {
+        return true;
+    }
+
+    @Override
+    public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connection connection,
+            JdbcTypeHandle typeHandle)
+    {
+        String jdbcTypeName = typeHandle.getJdbcTypeName()
+                .orElseThrow(() -> new TrinoException(JDBC_ERROR, "Type name is missing: " + typeHandle));
+
+        Optional<ColumnMapping> mapping = getForcedMappingToVarchar(typeHandle);
+        if (mapping.isPresent()) {
+            return mapping;
+        }
+        if (jdbcTypeName.equals("UInt8")) {
+            return Optional.of(booleanColumnMapping());
+        }
+        if (jdbcTypeName.equals("Int8")) {
+            return Optional.of(tinyintColumnMapping());
+        }
+        if (jdbcTypeName.equals("Int16")) {
+            return Optional.of(smallintColumnMapping());
+        }
+        if (jdbcTypeName.equals("Int32")) {
+            return Optional.of(integerColumnMapping());
+        }
+        if (jdbcTypeName.equals("Int64") || jdbcTypeName.equals("bigint")) {
+            return Optional.of(bigintColumnMapping());
+        }
+        if (jdbcTypeName.equals("Float32")) {
+            return Optional.of(realColumnMapping());
+        }
+        if (jdbcTypeName.equals("Float64")) {
+            return Optional.of(doubleColumnMapping());
+        }
+        if (typeHandle.getJdbcType() == Types.DECIMAL) {
+            int precision = min(typeHandle.getColumnSize().orElse(0), Decimals.MAX_PRECISION);
```

Comment:
can column size be missing? when?

can column size be > 38? if so, see OracleClient for how this should be handled
(ideally as a follow up PR, to keep scope of this under control)

Reply:
Yes, CK's decimal size > 38

Reply:
https://github.com/trinodb/trino/issues/6678